lib/galaxy/tools/parameters/basic.py

```diff
@@ -676,36 +676,19 @@ def get_initial_value( self, trans, context, history=None ):
 ## it for now to avoid breaking any tools.
 
 
-class BaseURLToolParameter( ToolParameter ):
+class BaseURLToolParameter( HiddenToolParameter ):
     """
     Returns a parameter the contains its value prepended by the
     current server base url. Used in all redirects.
     """
-
-    def __init__( self, tool, input_source ):
-        input_source = ensure_input_source( input_source )
-        ToolParameter.__init__( self, tool, input_source )
-        self.value = input_source.get( 'value', '' )
-
-    def get_value( self, trans ):
-        # url = trans.request.base + self.value
-        url = url_for( self.value, qualified=True )
-        return url
-
-    def get_html_field( self, trans=None, value=None, other_values={} ):
-        return form_builder.HiddenField( self.name, self.get_value( trans ) )
-
     def get_initial_value( self, trans, context, history=None ):
-        return self.value
+        self.url_for = url_for( '', qualified=True )
```

Comment:
We shouldn't generate this client-side, because then a person can have it set to whatever value that when by haxin the client code. It needs to come from the server.